src/leap/common/http.py

```diff
@@ -66,7 +186,7 @@ class HTTPClient(object):
     in order to avoid resource abuse on huge requests batches.
     """
 
-    def __init__(self, cert_file=None, pool=_pool):
+    def __init__(self, cert_file=None, pool=None, timeout=_defaultTimeout):
```

Comment:
I'm happy with having one pool per-HTTPClient, could we remove the global `_pool` in this case? And if we don't have a global pool I don't see the point on having the `pool` param in the constructor, we could remove it as well.